### PR TITLE
feat(devtools): merge mountpoint into client package

### PR DESCRIPTION
**Pull request by Asuka109**
### <samp>🤖 Generated by Copilot at 5e78e55</samp>

This pull request enhances the DevTools client package with TypeScript, ESLint, Radix UI, and dynamic mounting features. It updates the ESLint configuration, the Modern.js app configuration, and the `package.json` file. It also renames and moves some files to the `client` subfolder, and adds a new `mount.mjs` file that exports a function to inject the DevTools UI into any web page.

### <samp>🤖 Generated by Copilot at 5e78e55</samp>

*  Add a function `mountDevTools` to dynamically inject the DevTools UI into any web page ([link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-f35758d12a9cab28719ff57517f6b9c4cb66b381a8770b2cdf7a285ba8ab913aR1-R45))
*  Add a type annotation for the ESLint configuration object and disable the `curly` rule ([link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-402655299dffd8bd2a71146cbf98274288ccb45c5df7e50170452e28d58077fbR1), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-402655299dffd8bd2a71146cbf98274288ccb45c5df7e50170452e28d58077fbR10-R12))
*  Update the Modern.js app configuration to use TypeScript, change the entry points, and add a custom bundler chain ([link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-249610e4c3d6bd8dbb483f2667c5093d38461da73b5a217a63a55cbe74e0f2a3L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-249610e4c3d6bd8dbb483f2667c5093d38461da73b5a217a63a55cbe74e0f2a3L17-R31), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-249610e4c3d6bd8dbb483f2667c5093d38461da73b5a217a63a55cbe74e0f2a3R41-R48))
*  Update the `package.json` file of the DevTools client package to specify the entry points for the package ([link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-ee0bdc3f92472c79376377721c0e8c205228ee40cc94ffd50d4feb71deba0c2eR20-R25))
*  Rename the `radix-token-transformer.js` file and move it to the `plugins` folder ([link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-43455bc4ff50cb62c2d8812c2dcd06ba19fff54aa4011f45d972531b99aecddb))
*  Rename and move the files in the `src/components` folder to the `src/client/components` folder ([link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-70515c1273ccd6d8b03952167104e17c3918042d796fc2368808bc2c2f290f33), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-282b0691a02f385943d100a900a42a96fb02eb6b6d6ad77e0da6bf95171ed0dd), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-917abe8cfc5a35bd9713212b604bb9905960cd26d9325e7410ea23f847e7e4f8), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-59d2b45b7d9ae2615bb3ab6041313cebc82edf8f78e7c53f42e2f687495b9bd3), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-262f3e3f123926df5f2e07257aba64fb2ab433dc546cb1060e4da2ad2e57274f), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-44763d2f8aac686a21551f6cab56150372a6c42421614739e700ec6786b9b1a6), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-caf8f02d8ad93a8b75bbc545a5c00a3f487bfec9c127bc3878747c3af500ef25), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-0d5b4bc31d9f2bb402c37c439d6d4821c146fb07c98c165a4f5a977c74cfafa5), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-2948108acfc34262c08ac972ed3b3f8e518c777682ab20527728ae8d5b211c1d), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-84ec2331546b4b3d52c533e137ca0eb8ebe01597f275e37e8a710050a06e4cf5), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-fa5291b9f056c9b15551055d18bf0663efe6625b730e33581f733fc836827a5b), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-ac3e1f29660c9b867e246341b58c23df04bf5a00b73e356a60c05f9107beac04), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-128010a8dc598837358cea73a30c8c1df6e7df1a28610127027b73abb60d0435), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-f18da7aec518957d67c7d75c575ad48d6a5a720f9bc751134f70458e763fa4aa), [link](https://github.com/web-infra-dev/modern.js/pull/4920/files?diff=unified&w=0#diff-5ce5c295e9ab457436d7804e901a1a148890e1e8ce11c06dcb99eb2d2da3ad80))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by Asuka109** [comment]
!canary

**Comment by Asuka109** [comment]
!canary

File: packages/devtools/client/exports/mount.mjs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { parseQuery } from 'ufo';
```
Comment (chenjiahan) [inline packages/devtools/client/exports/mount.mjs:1]: Should we add this module to `source.include`?

DevTools may not work in legacy browsers.

Comment (Asuka109) [reply packages/devtools/client/exports/mount.mjs:1]: okk, I will do it.

Comment (Asuka109) [reply packages/devtools/client/exports/mount.mjs]: I refactor mount.mjs in es5. Now it should work well with legacy browsers.

Comment (chenjiahan) [reply packages/devtools/client/exports/mount.mjs]: Ok, we can add a compile step in the future because maintain es5 code is hard.

**Review by chenjiahan — APPROVED** [review]

File: packages/devtools/client/exports/mount.mjs (hunk 1 above)
Comment (Asuka109) [reply packages/devtools/client/exports/mount.mjs]: Sure.

**Pull request merged**